pkg/loadbalancer/redirectpolicy/controller.go

```diff
@@ -283,56 +308,43 @@ func (c *lrpController) processRedirectPolicy(wtxn writer.WriteTxn, lrpID lb.Ser
 		// In address-based mode there is no existing service/frontend to match against and
 		// instead the frontend is created here.
 		for _, feM := range lrp.FrontendMappings {
-			fe, _, found := c.p.Writer.Frontends().Get(wtxn, lb.FrontendByAddress(feM.feAddr))
-			if found {
-				if fe.Type != lb.SVCTypeLocalRedirect {
-					c.p.Log.Error("LocalRedirectPolicy matches an address owned by an existing service => refusing to override",
+			if len(pods) == 0 {
+				// No pods exist to redirect the traffic to. Remove the frontend to let the traffic
+				// be handled normally.
+				c.p.Writer.DeleteFrontend(wtxn, feM.feAddr)
```

Comment:
>c.p.Writer.DeleteFrontend

Does that mean if a policy selected backend pod(s) are deployed later, we would end up missing that update? Trying to understand why we need to delete the frontend here instead of not creating a local redirect service entry.

Reply:
This is recomputed if any of the inputs change, so if pods in the target namespace change this will get recomputed and the frontend is created.

We might have created the frontend earlier when we still had backend pods so we must delete it here as otherwise we'll blackhole the traffic. We could also delete the LRP service, but IMO it's clearer that we have it around as long as there's a policy for it and it'd just not have any frontends or backends associated to it if no local pods exist.